### ISSUE TITLE
Support for IPv6 networking

### ISSUE THREAD

**Issue by varbanv** [kyma-project/kyma#18724]
**Description**

Provide support for IPv6 based networking in Kyma clusters.

**Reasons**

Government customers are mandated to switch over to IPv6, some cloud providers are starting to charge a premium for IPv4 addresses, and some use cases can be better supported over IPv6. Thus we need no provide !Pv6 support for compliance, cost savings, and architectural reasons.

**Attachments**

Currently blocked by missing dual stack support in Gardener: [IPv6 Single-Stack Support in Local Gardener](https://github.com/gardener/gardener/blob/98fb871eb6045f69715d9f8b39dd84f9637f1fa0/docs/proposals/21-ipv6-singlestack-local.md#non-goals)

**Comment by anishj0shi** [comment]
IPv6 tests on GKE and Gardener: https://github.tools.sap/I322872/kyma-ipv6

on Gardener, the Poc is with ipv6 dual stack ingress only, because at the moment doesn't support kubernetes overlay network in dual stack.

**Comment by tobiscr** [comment]
POC was successfully finished. We need also adjustments on ISTIO side to get dual-stack supported.

@strekm : can you please provide the issue on your side and link it in this issue? THX in advance!

**Comment by tobiscr** [comment]
Closing this issue, will be continued in separated EPICs:

1. Support IPv6 in service-mesh (https://github.com/kyma-project/istio/issues/1598)
2. Enable customer facing Kyma modules to communicate with IPv6 (interal-backlog/#8317)
